src/global/global_init.cc

```diff
@@ -353,8 +353,9 @@ int global_init_prefork(CephContext *cct)
   const md_config_t *conf = cct->_conf;
   if (!conf->daemonize) {
 
-    if (pidfile_write(conf) < 0)
-      exit(1);
+    if (!(cct->get_init_flags() & CINIT_FLAG_NO_PIDFILES))
+      if (pidfile_write(conf) < 0)
+        exit(1);
 
     if ((cct->get_init_flags() & CINIT_FLAG_DEFER_DROP_PRIVILEGES) &&
```

Comment:
Maybe add the guard here also to prevent getting warning messages.  Not a huge deal as the return code for `chown_path()` isn't actually checked, so it wouldn't cause a failure.

What do you think?